composables/drop/useGenerativeDropMint.ts

```diff
@@ -50,30 +45,35 @@ export function useCollectionEntity(collectionId?: string) {
         : [() => drop.value?.collection, accountId],
     },
   )
+}
 
-  const maxCount = computed(() => collectionData.value?.collectionEntity?.max)
-
-  const mintedAmountForCurrentUser = computed(
-    () => collectionData.value?.nftEntitiesConnection?.totalCount ?? 0,
-  )
+export function useCollectionEntity(collectionId?: string) {
+  const { drop } = useDrop()
+  const { client } = usePrefix()
 
-  const description = computed(
-    () => collectionData.value?.collectionEntity?.meta?.description ?? '',
-  )
-  const collectionName = computed(
-    () => collectionData.value?.collectionEntity?.name ?? '',
-  )
+  const collectionKey = computed(() => collectionId ?? drop.value?.collection)
 
-  const nftCount = computed(
-    () => collectionData.value?.collectionEntity?.nftCount ?? 0,
+  const { data: collectionData } = useCollectionData(
+    collectionKey,
+    client,
+    collectionId,
+    drop,
   )
 
   return {
-    maxCount,
-    mintedAmountForCurrentUser,
-    description,
-    collectionName,
-    nftCount,
+    maxCount: computed(() => collectionData.value?.collectionEntity?.max),
+    mintedAmountForCurrentUser: computed(
+      () => collectionData.value?.nftEntitiesConnection?.totalCount ?? 0,
+    ),
+    description: computed(
+      () => collectionData.value?.collectionEntity?.meta?.description ?? '',
+    ),
+    collectionName: computed(
+      () => collectionData.value?.collectionEntity?.name ?? '',
+    ),
+    nftCount: computed(
+      () => collectionData.value?.collectionEntity?.nftCount ?? 0,
+    ),
   }
```

Comment:
IMO, the previous pattern is much better because we can clearly see all the exported functions. However, if it's just for fixing the codeclimate, let's do it.

Reply:
fixed
resolving